AUTHORING_GUIDE.md

```diff
@@ -1,207 +1,117 @@
-
-In general our sample format follows ideas borrowed from
-[Literate Programming](https://en.wikipedia.org/wiki/Literate_programming).
-Notably, your sample program should be self-contained, readable from top to bottom,
-and fairly self-documenting. Prefer descriptive names. Use comments
-and docstrings only as needed to further explain. Always introduce functions and
-variables before they are used. Prefer less indirection. Prefer imperative
-programming as it is easier to understand.
-
-### Shebang
+We're happy you want to write a Python sample! Like a lot of Pythonistas,
+we're opinationed and fussy. This guide is a reference for the format and
+style expected of samples contributed to the
+[python-docs-samples](https://github.com/GoogleCloudPlatform/python-docs-samples)
+repo. The guidelines below are intended to ensure that all Python samples
+meet the following goals:
+
+* **Copy-paste-runnable.** A developer should be able to copy and paste the
+code into their own environment and run it with as few modifications as
+possible.
+* **Teach through code.** Each sample should demonstrate best practices for
+interacting with Google Cloud libraries, APIs, or services.
+* **Idiomatic.** Each sample should follow widely accepted Python best
+practices as covered below.
+
+# Sample Guidelines
+
+This section covers guidelines for Python samples. Note that
+[Testing Guidelines](#testing-guidelines) are covered separately below.
+
+## Folder Location
+
+Each sample should be in a folder under the top-level folder of
+[python-docs-samples](https://github.com/GoogleCloudPlatform/python-docs-samples)
+that corresponds to the Google Cloud service or API used by the sample.
+For example, a sample demonstrating how to work with BigTable should be
+in a subfolder under the
+[python-docs-samples/bigtable](https://github.com/GoogleCloudPlatform/python-docs-samples/tree/master/bigtable)
+folder.
+
+Conceptually related samples under a service or API should be grouped into
+a subfolder. For example, App Engine Standard samples are under the
+[appengine/standard](https://github.com/GoogleCloudPlatform/python-docs-samples/tree/master/appengine/standard)
+folder, and App Engine Flex samples are under the
+[appengine/flexible](https://github.com/GoogleCloudPlatform/python-docs-samples/tree/master/appengine/flexible)
+folder.
+
+If your sample is a set of discrete code snippets that each demonstrate a
+single operation, these should be grouped into a `snippets` folder. For
+example, see the snippets in the
+[bigtable/snippets/writes](https://github.com/GoogleCloudPlatform/python-docs-samples/tree/master/bigtable/snippets/writes)
+folder.
+
+If your sample is a quickstart — intended to demonstrate how to quickly get
+started with using a service or API — it should be in a _quickstart_ folder.
+
+## Python Version
+
+As of this writing (early 2020), the default Python version for samples is
+Python 3.6. All samples should pass tests on 3.6, and support for 3.7 and
```

Comment:
I would be inclined to say it's 3.7 at this point - but what do other folks think?

Reply:
I think we have a couple different options. that there probably needs to be some more discussion around (for new versions and significant changes):
1. Target the latest stable python version 
2. Target the current most popular python version (and adjust every ~6 months)?
3. Target the earliest version the client libraries support

Reply:
I've updated to say 3.7 for now, and I agree that we need to discuss/decide a general strategy for Python versions going forward. Addressed in commit [751c8e2](https://github.com/GoogleCloudPlatform/python-docs-samples/pull/2728/commits/751c8e27c591a489d2cb47477b0a971d41fcf36a)